### DISCUSSION TITLE
My dApp runs on noble-hashes 1.1 and noble-secp256k1 1.5.5, without web3.js! (almost)

### DISCUSSION THREAD

**Discussion by ghost** [paulmillr/noble-hashes#34] [Q&A] [answered]
I have an ethereum based dApp that relies on web3.js only for transactions and for encoding/decoding of web3 wallets. 

I got almost entirely rid of web3.js saving a lot of kB of code to download and my dApp starts way quicker than before, all by using noble libraries.

Thank you for that! Some parts of my dApp are roughly 8x faster than before. The noble code was easy to integrate and substituting legacy web3 code was straightforward. Thank you for getting rid of BN and Buffer, btw.

But for one purpose I have to use the legacy cipher/decipher of web3.js: Encoding and decoding of wallets. (for being compatible with web3.js, metamask, MyCrypto and others)

From my understanding, using noble shake128/shake256 combined with a noble kdf works like a cypher, comparable to aes - cyphers and I am using this successfully in my dApp.

Am I right that shake-based cyphers cannot be backwards compatible with e.g aes-128-ctr based cyphers?
Or is there anything I am missing here?

regards,
Peter

P.S.: This is NOT a feature request. :)     This is for my understanding.

**Comment by paulmillr** [comment, marked as answer]
Yes, correct. 

Also, take a look at aes-js: it also has 0 dependencies https://www.npmjs.com/package/aes-js

**Comment by ghost** [comment]
thank you
